contracts/Whitelist.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.6.10;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @author Opyn Team
+ * @title Whitelist Module
+ * @notice The whitelist module keeps track of all valid Otoken contracts.
+ */
+contract Whitelist is Ownable {
+    mapping(bytes32 => bool) internal isSupportedProduct;
+
+    event ProductWhitelisted(
+        bytes32 productHash,
+        address indexed underlying,
+        address indexed collateral,
+        address indexed strike
+    );
+
+    /**
+     * @notice check if a product is supported
+     * @dev product = the hash of underlying, collateral and strike asset
+     * @param _underlying option underlying asset address
+     * @param _collateral option collateral asset address
+     * @param _underlying option strike asset address
+     * @return boolean, true if product is supported
+     */
+    function isProductSupported(
+        address _underlying,
+        address _strike,
+        address _collateral
+    ) external view returns (bool) {
+        bytes32 productHash = keccak256(abi.encode(_underlying, _collateral, _strike));
```

Comment:
We should unify the order of these 3 assets to underlying, strike, collateral everywhere.

Reply:
in the spec it is underlying, collateral and strike, in that order.

Reply:
Okay we need to make that underlying, strike, collateral. I asked Andrew and he said that makes more sense.